protostar/cheatable_starknet/cheatables/cheatable_syscall_handler.py

```diff
@@ -31,31 +41,68 @@ def __init__(self, message: str):
 
 
 class CheatableSysCallHandler(BusinessLogicSysCallHandler):
-    def __init__(self, state: StateSyncifier, **kwargs: Any):
-        self.cheatable_state: "CheatableCachedState" = cast(
-            "CheatableCachedState", state.async_state
-        )
+    block_info_controller_state: Optional["BlockInfoControllerState"] = None
+    contracts_controller_state: Optional["ContractsControllerState"] = None
```

Comment:
This is not a dataclass, which makes those class variables

Reply:
Resolving in favor of https://github.com/software-mansion/protostar/pull/1451#discussion_r1101478245